### ISSUE TITLE
Metadata missing after image upload

### ISSUE THREAD

**Issue by keinkurt** [Piwigo/Piwigo-Mobile#209]
If I upload an image (live or not live) all metadata (creation date, gps, orientation, ...) are lost at piwigo. If I upload an image via Web Interface it is working as expected.

Mobile: iPhone SE with iOS 11.2.1
Server:
* Piwigo 2.9.2
* Operating system: Linux
* PHP: 7.0.25-1+0~20171027135825.8+stretch~1.gbp75b787 (Show info) [2018-01-03 10:04:04]
* MySQL: 5.5.5-10.1.29-MariaDB-6 [2018-01-03 10:04:04]
* Graphics Library: GD 2.2.4

**Comment by EddyLB** [comment]
I have taken a photo and a live photo with my iPhone and uploaded them to a Piwigo. All metadata are preserved as expected. Are you sure you are using the version 2.1.5 of Piwigo iOS ? Can you please check that the "Strip Private Metadata" flag is unchecked in the Settings page ?

**Comment by keinkurt** [comment]
Yes, app version is 2.1.5 (79) and in my settings "Strip Private Metadata" is unchecked.

**Comment by keinkurt** [comment]
I have found this in a log file on server `_data/logs/log_2018-01-03_ff9d87b47eec94ec04a59a672cb87803a9bb99ad.txt`:

```
[2018-01-03 11:37:42]   [INFO]  upload_file_pdf, $file_path = ././upload/2018/01/03/20180103113742-76d826af.jpg, $representative_ext =
[2018-01-03 11:37:42]   [INFO]  upload_file_tiff, $file_path = ././upload/2018/01/03/20180103113742-76d826af.jpg, $representative_ext =
[2018-01-03 11:37:42]   [INFO]  upload_file_video, $file_path = ././upload/2018/01/03/20180103113742-76d826af.jpg, $representative_ext =
[2018-01-03 11:37:42]   [INFO]  Handling ././upload/2018/01/03/20180103113742-76d826af.jpg got
[2018-01-03 11:37:43]   [DEBUG] [i.php]
  src_path: '20180103113742-76d826af.jpg'
  derivative_path: '20180103113742-76d826af-th.jpg'
  o_size: '4032 3024 12192768'
  d_size: '144 108 15552'
  mem_usage: 0.40000000000000002
  timing: array(
    'load' => 140,
    'rotate' => '',
    'crop' => '',
    'scale' => 171,
    'sharpen' => '',
    'watermark' => '',
    'save' => 2,
    'send' => 0,
    'total' => 314,
  )
```

Maybe it can help...

**Comment by EddyLB** [comment]
This is a standard log showing that the image has been recognised as a JPG file by the server.
Could you please check that Photos and Camera are allowed to use location metadata (in Settings / Privacy / Location Services) in your iPhone ?

**Comment by keinkurt** [comment]
I have checked all. Piwigo has rights to read and write Photos and Camera has rights to get location services.

**Comment by EddyLB** [comment]
Can you please contact [me](mailto:piwigo@lelievre-berna.net) so that I can give you credentials allowing you to perform a test with a freshly taken photo ? If you could give me a temporary access to an album of your Piwigo, that would also help.

**Comment by EddyLB** [comment]
Indeed, in some circumstances the metadata can be stripped before upload.
That will be fixed in release 2.1.6. Thank you for reporting and allowing me to test with your Piwigo.
